wiki/Ranking_Criteria/osu!catch/en.md

```diff
@@ -85,122 +46,142 @@ Overall rules and guidelines apply to every kind of osu!catch difficulty. Rhythm
+- **[Hyperdashes](/wiki/Glossary/Hyperdash) that are [higher-snapped](/wiki/Glossary/Snapping#higher-snapped) should not be followed by [antiflow](/wiki/Beatmapping/Antiflow) [dashes](/wiki/Glossary/Dash) with a gap lower than 250ms.**
+- **[Edge dashes](/wiki/Glossary/Edge_dash) may only be used singularly (not in conjunction with other [dashes](/wiki/Glossary/Dash) or [hyperdashes](/wiki/Glossary/Hyperdash)).**
+- **[Combos](/wiki/Beatmapping/Combo) should not exceed 16 objects including slider tails and repeats.** [Spinners](/wiki/Hit_object/Spinner) are an exception.
 - **Note density should follow a mostly 1/2 + 1/4 and/or 1/3 + 1/6 pattern.** 1/8 patterns and higher should be used sparingly.
 
 #### Difficulty setting guidelines
 
-- Approach Rate / Overall Difficulty should be between 7 and 9.
-- HP Drain Rate should be between 5 and 6.
-- Circle Size should be between 3 and 5.
+- [Approach Rate](/wiki/Beatmapping/Approach_rate) / [Overall Difficulty](/wiki/Beatmapping/Overall_difficulty) should be between 7 and 9.
+- [HP Drain Rate](/wiki/Beatmapping/HP_drain_rate) should be between 5 and 6.
+- [Circle Size](/wiki/Beatmapping/Circle_size) should be between 3 and 5.
 
 ### ![](/wiki/shared/diff/expert-c.png) Overdose
 
 #### Rules
 
-- **At least 62 ms must be left between circles/sliders and the start of spinners.** This is to ensure readability.
-- **At least 125 ms must be left between circles/sliders and the end of spinners.** This is to ensure readability.
+- **At least 62 ms must be left between circles/sliders and the start of [spinners](/wiki/Hit_object/Spinner).** This is to ensure readability.
+- **At least 125 ms must be left between circles/sliders and the end of [spinners](/wiki/Hit_object/Spinner).** This is to ensure readability.
 
 #### Guidelines
 
-- **Hyperdashes should only be used on drops and/or slider repetitions when the slider path is simple and easy-to-follow.** This is to prevent chaotic or unreasonably difficult slider movement, as transitioning into and out of complex slider shapes with hyperdash is usually uncomfortable to play and a major penalty to accuracy for little benefit.
-- **1/8 and higher hyperdashes should not be used between consecutive object pairs.**
-- **Edge dashes may be used with caution for a maximum of three consecutive objects, and should not be used after hyperdashes.**
-- **Combos should not exceed 16 objects including slider tails and repeats.** Spinners are an exception.
+- **[Hyperdashes](/wiki/Glossary/Hyperdash) should only be used on [drops](/wiki/Hit_object/Juice_stream#drop) and/or slider repetitions when the slider path is simple and easy-to-follow.** This is to prevent chaotic or unreasonably difficult slider movement, as transitioning into and out of complex slider shapes with [hyperdash](/wiki/Glossary/Hyperdash) is usually uncomfortable to play, and a major penalty to accuracy for little benefit.
```

Comment:
```suggestion
- **[Hyperdashes](/wiki/Glossary/Hyperdash) should only be used on [drops](/wiki/Hit_object/Juice_stream#drop) and/or slider repetitions when the slider path is simple and easy-to-follow.** This is to prevent chaotic or unreasonably difficult slider movement, as transitioning into and out of complex slider shapes with hyperdash is usually uncomfortable to play, and a major penalty to accuracy for little benefit.
```